contracts/contracts/levels/RecoveryFactory.sol

```diff
@@ -7,25 +7,26 @@ import './Recovery.sol';
 
 contract RecoveryFactory is Level {
 
-  address lostAddress;
+  mapping (address => address) lostAddress;
 
-  function createInstance(address) override public payable returns (address) {
+  function createInstance(address _player) override public payable returns (address) {
     Recovery recoveryInstance;
     recoveryInstance = new Recovery();
     // create a simple token 
     recoveryInstance.generateToken("InitialToken", uint(100000));
     // the lost address
-    lostAddress = address(uint160(uint256(keccak256(abi.encodePacked(uint8(0xd6), uint8(0x94), recoveryInstance, uint8(0x01))))));
+    lostAddress[_player] = address(uint160(uint256(keccak256(abi.encodePacked(uint8(0xd6), uint8(0x94), recoveryInstance, uint8(0x01))))));
     // Send it some ether
-    (bool result,) = lostAddress.call{value:0.5 ether}("");
+    (bool result,) = lostAddress[_player].call{value:0.5 ether}("");
     require(result);
 
     return address(recoveryInstance);
   }
 
-  function validateInstance(address payable _instance, address) override public returns (bool) {
+  function validateInstance(address payable _instance, address _player) override public returns (bool) {
     require(_instance != address(0)); // Suppress solidity warning. 
-    if (address(lostAddress).balance == 0) {
+    if (address(lostAddress[_player]).balance == 0) {
+      delete lostAddress[_player];
       return true;
     }
     return false;
```

Comment:
Looking at other factories, `validateInstance` remains working after the level is won, so we shouldn't `delete` the lost address.

```suggestion
    return address(lostAddress[_player]).balance == 0;
```